source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1034,15 +1034,14 @@ ShaderNode* ShaderGraph::createNode(const Node& node, GenContext& context)
         if (!interfaceName.empty())
         {
             ShaderGraphInputSocket* inputSocket = getInputSocket(interfaceName);
-            if (!inputSocket)
+            if (inputSocket)
```

Comment:
Out of curiosity, in what context is this no longer an error (to have an input on a node inside a graph that is specifying a connection to something that does not exists)?

Reply:
Having the input on a node have an "interfacename" without the "nodegraph" being specified as a "nodedef". So to support the workflow of creating the interface but not necessarily creating a new definition. For us, we've also created the child "input" to expose the interface for the nodegraph so it is considered "connected" to the interface.

Reply:
Ok, I'm fine with removing the error from here since this might be better placed in the validation code section.

I guess in the future we would have the input created explicitly on the nodegraph, no matter if it has a nodedef of not.